docs/core/compatibility/index.md

```diff
@@ -183,9 +183,13 @@ Changes in this category modify the public surface area of a type. Most of the c
 
    The return value of a method or the type of a property or field cannot be modified. For example, the signature of a method that returns an <xref:System.Object> cannot be changed to return a <xref:System.String>, or vice versa.
 
-- ❌ **DISALLOWED: Adding a field to a struct that previously had no state**
+- ❌ **DISALLOWED: Adding an instance field to a struct that has no nonpublic fields**
 
-  Definite assignment rules allow the use of uninitialized variables so long as the variable type is a stateless struct. If the struct is made stateful, code could end up with uninitialized data. This is both potentially a source breaking and a binary breaking change.
+  If a struct has only public fields or has no fields at all, callers can declare locals of that struct type without calling the struct's ctor or first initializing the local to _default(T)_, so long as all public fields are set on the struct before first use. Adding any new fields - public or nonpublic - to such a struct is a source breaking change for these callers, as the compiler will now require the additional fields to be initialized.
+
+  Additionally, adding any new fields - public or nonpublic - to such a struct is a binary breaking change to callers who have applied `[SkipLocalsInit]` to their code. Since the compiler wasn't aware of these fields at compile time, it could emit IL which does not fully initialize the struct, leading to the struct being created from uninitialized stack data.
```

Comment:
```suggestion
  Additionally, adding any new fields - public or nonpublic - to a struct with no fields or only public fields is a binary breaking change to callers who've applied `[SkipLocalsInit]` to their code. Since the compiler wasn't aware of these fields at compile time, it could emit IL that doesn't fully initialize the struct, leading to the struct being created from uninitialized stack data.
```